usvm-ts/src/main/kotlin/org/usvm/TSInterpreter.kt

```diff
@@ -40,14 +57,117 @@ class TSInterpreter(
+            is EtsThrowStmt -> visitThrowStmt(scope, stmt)
+            is EtsGotoStmt -> visitGotoStmt(scope, stmt)
+            is EtsNopStmt -> visitNopStmt(scope, stmt)
+            is EtsSwitchStmt -> visitSwitchStmt(scope, stmt)
+            else -> error("Unknown stmt: $stmt")
         }
 
         return scope.stepResult()
     }
 
+    private fun visitIfStmt(scope: TSStepScope, stmt: EtsIfStmt) {
+        val exprResolver = exprResolverWithScope(scope)
+
+        val boolExpr = exprResolver
+            .resolveTSExpr(stmt.condition)
+            ?.asExpr(ctx.boolSort)
+            ?: return
+
+        val succs = applicationGraph.successors(stmt).take(2).toList()
+        val (posStmt, negStmt) = succs[0] to succs[1]
+
+        scope.forkWithBlackList(
+            boolExpr,
+            posStmt,
+            negStmt,
+            blockOnTrueState = { newStmt(posStmt) },
+            blockOnFalseState = { newStmt(negStmt) }
+        )
+    }
+
+    private fun visitReturnStmt(scope: TSStepScope, stmt: EtsReturnStmt) {
+        val exprResolver = exprResolverWithScope(scope)
+
+        val method = requireNotNull(scope.calcOnState { callStack.lastMethod() })
+        val returnType = method.returnType
+
+        val valueToReturn = stmt.returnValue
+            ?.let { exprResolver.resolveTSExpr(it, returnType) ?: return }
+            ?: ctx.mkUndefinedValue()
+
+        scope.doWithState {
+            returnValue(valueToReturn)
+        }
+    }
+
+    private fun visitAssignStmt(scope: TSStepScope, stmt: EtsAssignStmt) {
+        val exprResolver = exprResolverWithScope(scope)
+
+        val lvalue = exprResolver.resolveLValue(stmt.lhv) ?: return
+        val expr = exprResolver.resolveTSExpr(stmt.rhv, stmt.lhv.type) ?: return
+
+        scope.doWithState {
+            memory.write(lvalue, expr)
+            val nextStmt = stmt.nextStmt ?: return@doWithState
+            newStmt(nextStmt)
+        }
+    }
+
+    private fun visitCallStmt(scope: TSStepScope, stmt: EtsCallStmt) {
+        TODO()
+    }
+
+    private fun visitThrowStmt(scope: TSStepScope, stmt: EtsThrowStmt) {
+        TODO()
+    }
+
+    private fun visitGotoStmt(scope: TSStepScope, stmt: EtsGotoStmt) {
+        TODO()
+    }
+
+    private fun visitNopStmt(scope: TSStepScope, stmt: EtsNopStmt) {
+        TODO()
+    }
+
+    private fun visitSwitchStmt(scope: TSStepScope, stmt: EtsSwitchStmt) {
+        TODO()
+    }
+
+    private fun exprResolverWithScope(scope: TSStepScope) =
+        TSExprResolver(
+            ctx,
+            scope,
+            ::mapLocalToIdxMapper,
+        )
+
+    // (method, localName) -> idx
+    private val localVarToIdx = mutableMapOf<EtsMethod, MutableMap<String, Int>>()
+
+    private fun mapLocalToIdxMapper(method: EtsMethod, local: EtsValue) =
+        when (local) {
+            is EtsLocal -> {
+//                // See https://github.com/UnitTestBot/usvm/issues/202
+//                if (local.name == "this") 0 else
```

Comment:
Is it fixed or is it not fixed? Also, we should not create a commented code in a regular scenarios. Moreover, create comments not with "see", but with "TODO", because there is no fast way to check this sees